src/wallet/wallet_tx_builder.cpp

```diff
@@ -21,6 +23,9 @@ static size_t PadCount(size_t n)
 
 static CAmount
 CalcZIP317Fee(
+        const Consensus::Params& consensus,
+        const CWallet& wallet,
+        const CChain& chain,
```

Comment:
Consider passing `consensusBranchId` through each layer, instead of `consensus` and `chain`. I believe this ends up simpler: you still only have to compute the consensus branch ID once (around line 422 I think), and it passes fewer parameters while clarifying what information is actually needed.